components/playlist/browser/resources/components/playlistFolder.tsx

```diff
@@ -114,11 +115,38 @@ function EditActionsContainer ({
   )
 }
 
+function useItemIdFromHash () {
+  const getId = () => window.location.hash.replace('#', '')
+  let [idFromHash, setIdFromHash] = React.useState<string>('')
+  const updateId = () => setIdFromHash(getId())
+
+  React.useEffect(() => {
+    updateId()
+    window.addEventListener('hashchange', updateId)
+    return () => window.removeEventListener('hashchange', updateId)
+  }, [])
+  return idFromHash
+}
+
+function useScrollToItem (itemId: string | undefined) {
+  const ref = React.useRef<HTMLAnchorElement>(null)
+  const scrollToElement = () => {
+    if (ref.current) {
+      window.scrollTo({ top: ref.current.offsetTop })
+    }
+  }
+
+  React.useEffect(() => scrollToElement(), [itemId, ref.current])
```

Comment:
So when `ref.current` changes, this effect won't necessarily be rerun (as changes to refs don't trigger rerenders). Fortunately, React refs are either a `Ref<T>` or a function `(ref: T) => void` so instead of returning a `ref` here, we should call a function which will be invoked whenever the referenced element changes:

```tsx
const useScrollToItem = (itemId: string | undefined) => {
    const [el, setEl] = useState<HTMLAnchorElement>(null)

   useEffect(() => {
        if (el) window.scrollTo({ top: el.offsetTop })
   }, [el, itemId])

   return setEl
}

```

Reply:
> So when ref.current changes, this effect won't necessarily be rerun (as changes to refs don't trigger rerenders).

Oh, so the effect is re-registered but doesn't need to be re run when `ref.current` changes? The more I hear about useEffect, the more it's like the footgun 😂

Reply:
Basically, changes to refs don't trigger rerenders (and if we don't rerender, the effect won't get run)